Define check_sizes_nonnegative as inline

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #71640

Moving this function into the cpp file caused a small regression in
empty_cpu's callgrind instruction count, so moving it back.

Differential Revision: [D33712880](https://our.internmc.facebook.com/intern/diff/D33712880)

peterbell10 committed:
Define check_sizes_nonnegative as inline

Moving this function into the cpp file caused a small regression in
empty_cpu's callgrind instruction count, so moving it back.

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33712880).

peterbell10 committed:
Rebase on working master on "Define check_sizes_nonnegative as inline"


Moving this function into the cpp file caused a small regression in
empty_cpu's callgrind instruction count, so moving it back.

Differential Revision: [D33712880](https://our.internmc.facebook.com/intern/diff/D33712880)

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33712880).
